Add Registry Documentation (#1977)
semgrep/semgrep-docs: docs/faq.md

Yes! Semgrep is safe to run on your private code. The [Semgrep Rules License v. 1.0](https://semgrep.dev/legal/rules-license/) restrictions only come into effect if you are **selling** a product using rules provided in the Semgrep Registry. If that’s the case, contact [<i class="fa-regular fa-envelope"></i> partnerships@semgrep.com](mailto:partnerships@semgrep.com) to learn more.

### How does the `semgrep/semgrep-rules` repository differ from the Semgrep Registry?

> The Semgrep Registry can import rules from various sources aside from the `semgrep/semgrep-rules` repository. Other sources include the [<i class="fas fa-external-link fa-xs"></i> Trail of Bits rules](https://github.com/trailofbits/semgrep-rules). These rules have their own licenses. 

Reviewer: ```suggestion
The Semgrep Registry can import rules from sources other than the `semgrep/semgrep-rules` repository, such as [<i class="fas fa-external-link fa-xs"></i> Trail of Bits](https://github.com/trailofbits/semgrep-rules). These rules have their own licenses. 
```
